imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -50,203 +47,207 @@ def hit_l1a(packet_file: str, data_version: str) -> list[xr.Dataset]:
-            #  leak_i_raw can be handled in the housekeeping class as an
-            #  InitVar so that it doesn't show up when you extract the object's
-            #  field names.
-        elif apid == HitAPID.HIT_SCIENCE:
-            logical_source = "imap_hit_l1a_sci-counts"
-            # TODO what about pulse height? It has the same apid.
-            #  Will need to approach this differently
-        else:
-            raise Exception(f"Unknown APID [{apid}]")
-        metadata_arrays = defaultdict(list)
-        for packet in data_packets:
-            # Add metadata to an array
-            for field in fields(packet):
-                field_name = field.name
-                field_value = getattr(packet, field_name)
-                # convert key to lower case to match SPDF requirement
-                data_key = field_name.lower()
-                metadata_arrays[data_key].append(field_value)
-
-        # Convert integers into datetime64[s]
-        epoch_converted_times = met_to_j2000ns(metadata_arrays["shcoarse"])
-
-        # Create xarray data arrays for dependencies
-        epoch_time = xr.DataArray(
-            epoch_converted_times,
-            name="epoch",
-            dims=["epoch"],
-            attrs=attr_mgr.get_variable_attributes("epoch"),
-        )
-
-        adc_channels = xr.DataArray(
-            np.arange(64, dtype=np.uint16),
-            name="adc_channels",
-            dims=["adc_channels"],
-            attrs=attr_mgr.get_variable_attributes("adc_channels"),
-        )
-
-        # NOTE: LABL_PTR_1 should be CDF_CHAR.
-        adc_channels_label = xr.DataArray(
-            adc_channels.values.astype(str),
-            name="adc_channels_label",
-            dims=["adc_channels_label"],
-            attrs=attr_mgr.get_variable_attributes("adc_channels_label"),
-        )
-
-        # Create xarray dataset
-        dataset = xr.Dataset(
-            coords={
-                "epoch": epoch_time,
-                "adc_channels": adc_channels,
-                "adc_channels_label": adc_channels_label,
-            },
-            attrs=attr_mgr.get_global_attributes(logical_source),
+    # Drop variables not needed for CDF
+    dataset = dataset.drop_vars(drop_keys)
+
+    # Create data arrays for dependencies
+    adc_channels = xr.DataArray(
+        np.arange(64, dtype=np.uint16),
+        name="adc_channels",
+        dims=["adc_channels"],
+        attrs=attr_mgr.get_variable_attributes("adc_channels"),
+    )
+
+    # NOTE: LABL_PTR_1 should be CDF_CHAR.
+    adc_channels_label = xr.DataArray(
+        adc_channels.values.astype(str),
+        name="adc_channels_label",
+        dims=["adc_channels_label"],
+        attrs=attr_mgr.get_variable_attributes("adc_channels_label"),
+    )
+
+    # Update dataset coordinates and attributes
+    dataset = dataset.assign_coords(
+        {
+            "adc_channels": adc_channels,
+            "adc_channels_label": adc_channels_label,
```

Comment:
I forget, do the "labels" need to be defined as coordinates as well, or are those variables that are dependent on the coordinates?

i.e. should `adc_channels_label` actually be in the `data_vars` section?

Reply:
I think it needs to be added to the coordinates as well because I initially didn't have it in there but then had to add it. I don't remember why though. I'll get clarification on this.

Reply:
I checked with Tenzin and she explained that cdflib updates requires this for data with greater than 1 dimensions

Reply:
Tenzin explained that it’s because cdflib updates requires labels in the coordinates for data with dimensions greater than 1